### ISSUE TITLE
[Bug]: replica cluster switchover doesn't work when only object store replication is configured

### ISSUE THREAD

**Issue by sujitkagarwal** [cloudnative-pg/cloudnative-pg#4772]
### Is there an existing issue already for this bug?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new bug.

### I have read the troubleshooting guide

- [X] I have read the troubleshooting guide and I think this is a new bug.

### I am running a supported version of CloudNativePG

- [X] I have read the troubleshooting guide and I think this is a new bug.

### Contact Details

sujit21agarwal@gmail.com

### Version

1.23.1

### What version of Kubernetes are you using?

1.27

### What is your Kubernetes environment?

Self-managed: kind (evaluation)

### How did you install the operator?

YAML manifest

### What happened?

I have created 2 cluster Primary and Replica.Replica cluster is created from object storage s3.

I have inserted data into primary cluster and data is successfully  replicated into the Replica Cluster.After that I followed below steps,

1. Updating the Primary Cluster  by making  
                    replica: 
                         enabled: true
2. Updating the Replica Cluster by making
                    replica: 
                         enabled: false.

I can see the status is changed from  Designated primary to Primary for replica cluster and also vice versa.

The issue which I am facing , when I inserted the data into new primary , data is not replicating into the replica cluster.

### Cluster resource

```shell
----Primary Cluster---

apiVersion: postgresql.cnpg.io/v1
kind: Cluster
metadata:
  name:  primary-cluster-example
  namespace: primary
spec:
  instances: 2
  storage:
    size: 1Gi
  enableSuperuserAccess: true
  superuserSecret:
    name: superuser-secret
  backup:
    barmanObjectStore:
      destinationPath: s3://backup/primary/
      endpointURL: http://seaweedfs-s3.locals3.svc.cluster.local:8333
      s3Credentials:
        accessKeyId:
          name: primary-cluster-example-secret-s3
          key: ACCESS_KEY_ID
        secretAccessKey:
          name: primary-cluster-example-secret-s3
          key: ACCESS_SECRET_KEY
      wal:
        compression: gzip
        encryption: AES256
        maxParallel: 8

  replica:
    enabled: false
    source: secondary-cluster-example



  externalClusters:
    - name: secondary-cluster-example
      barmanObjectStore:
        destinationPath: s3://backup/secondary/
        endpointURL: http://seaweedfs-s3.locals3.svc.cluster.local:8333
        s3Credentials:
          accessKeyId:
            name: primary-cluster-example-secret-s3
            key: ACCESS_KEY_ID
          secretAccessKey:
            name: primary-cluster-example-secret-s3
            key: ACCESS_SECRET_KEY


--Replica Cluster--------------------

apiVersion: postgresql.cnpg.io/v1
kind: Cluster
metadata:
  name:  secondary-cluster-example
  namespace: secondary
spec:
  instances: 2
  storage:
    size: 1Gi
  enableSuperuserAccess: true
  superuserSecret:
    name: superuser-secret
  backup:
    barmanObjectStore:
      destinationPath: s3://backup/secondary/
      endpointURL: http://seaweedfs-s3.locals3.svc.cluster.local:8333
      s3Credentials:
        accessKeyId:
          name: secondary-cluster-example-secret-s3
          key: ACCESS_KEY_ID
        secretAccessKey:
          name: secondary-cluster-example-secret-s3
          key: ACCESS_SECRET_KEY
      wal:
        compression: gzip
        encryption: AES256
        maxParallel: 8

  replica:
    enabled: true
    source: primary-cluster-example


  bootstrap:
    recovery:
      source: primary-cluster-example


  externalClusters:
    - name: primary-cluster-example
      barmanObjectStore:
        destinationPath: s3://backup/primary/
        endpointURL: http://seaweedfs-s3.locals3.svc.cluster.local:8333
        s3Credentials:
          accessKeyId:
            name: secondary-cluster-example-secret-s3
            key: ACCESS_KEY_ID
          secretAccessKey:
            name: secondary-cluster-example-secret-s3
            key: ACCESS_SECRET_KEY
```

### Relevant log output

```shell
no such logs I can see.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mnencia** [comment]
Hi @sujitkagarwal. Unfortunately, the demotion doesn't work at the moment when you are only using the object store to feed the replica cluster. This behavior is due to how PostgreSQL archives the last wal during the shutdown; see https://github.com/cloudnative-pg/cloudnative-pg/issues/4586 for the details. We are working on a fix.

**Comment by sujitkagarwal** [comment]
@mnencia  Do you know when new version will be released 1.24 with partial fix

**Comment by gbartolini** [comment]
Jul 16 - see: https://github.com/cloudnative-pg/cloudnative-pg/milestone/22
